build-tools/packages/build-cli/src/commands/generate/typetests.ts

```diff
@@ -0,0 +1,377 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import path from "node:path";
+import {
+	type BrokenCompatTypes,
+	type Logger,
+	type Package,
+	type PackageJson,
+	type TestCaseTypeData,
+	type TypeData,
+	buildTestCase,
+	getFullTypeName,
+	getNodeTypeData,
+	typeOnly,
+} from "@fluidframework/build-tools";
+import { Flags } from "@oclif/core";
+import { PackageName } from "@rushstack/node-core-library";
+import * as changeCase from "change-case";
+import { mkdirSync, readJson, rmSync, writeFileSync } from "fs-extra";
+import * as resolve from "resolve.exports";
+import { ModuleKind, ModuleResolutionKind, Project, type SourceFile } from "ts-morph";
+import { PackageCommand } from "../../BasePackageCommand";
+import { ApiLevel, ensureDevDependencyExists, knownApiLevels } from "../../library";
+import { unscopedPackageNameString } from "./entrypoints";
+
+export default class GenerateTypetestsCommand extends PackageCommand<
+	typeof GenerateTypetestsCommand
+> {
+	static readonly description = "Generates type tests for a package or group of packages.";
+
+	static readonly flags = {
+		level: Flags.string({
+			description: "What API level to generate tests for.",
+			default: ApiLevel.internal,
```

Comment:
My goal for this initial change was just to unblock type test generation on main since the original PR had been stalled for almost six months. Using ./internal was the most consistent with how the tests worked prior to API trimming. AB#7874 is a follow up item to figure out what combinations we want to test.